### ISSUE TITLE
Component - Add custom Zitadel IDP component to aspire

### ISSUE THREAD

**Issue by MadL1me** [CommunityToolkit/Aspire#350]
### .NET Aspire issue link

https://github.com/dotnet/aspire/issues/6988

### Overview

While the keycloak is widely used, there is some issues with it, especially in high load situations. There are a lot of different IDPs (Identitiy Providers), which can be used, with SDKs in dotnet as well (Auth0, LogTo, Authentic, etc)

The issues proposes adding component for one of the promising IDP, our company and many others using - [Zitadel](https://zitadel.com/).

### Usage example

Same as any other component

### Help us help you

Yes, I'd like to be assigned to work on this item

**Comment by aaronpowell** [comment]
> I am definitely keeping an eye on this issue. I made my own Aspire-Zitadel integration earlier but it would be nice to adopt a library with broader visibility and support rather than something homebrew.

Happy for you to contribute it here if you would like

**Comment by Foxtrek64** [comment]
> > I am definitely keeping an eye on this issue. I made my own Aspire-Zitadel integration earlier but it would be nice to adopt a library with broader visibility and support rather than something homebrew.
> 
> Happy for you to contribute it here if you would like

I should be able to get a PR in here soon, worst case by this weekend. Do I need to make a proposal champion first? Didn't see it in the contribution guide but it's possible I just missed it.

**Comment by aaronpowell** [comment]
No, linking to this issue is fine. Also, happy for you to join as a contributor and be the [CODEOWNER](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) of the integration

**Comment by aaronpowell** [comment]
No stress @Foxtrek64 - our stale issue monitor is more to ensure we don't have a massive backlog of stuff that never gets actioned.

**Comment by Foxtrek64** [comment]
PR #549 has been submitted, albeit currently as WIP. I lost my original code (doesn't seem to be anywhere in git and I'm usually pretty good about checking in code), so I've needed to recreate it from scratch. This is where the additional time to implement has come from.

The WIP PR is about 90% done - just need to add in database support, the health check, and get documentation and tests written. Otherwise we're just about good to go.
